engine/renderer/private/particles/particle_pipeline.cpp

```diff
@@ -87,92 +125,159 @@ void ParticlePipeline::RecordCommands(vk::CommandBuffer commandBuffer, uint32_t
+    // HDR color
+    colorAttachmentInfos[0].imageView = resources->ImageResourceManager().Access(_hdrTarget)->views[0];
+    colorAttachmentInfos[0].imageLayout = vk::ImageLayout::eAttachmentOptimalKHR;
+    colorAttachmentInfos[0].storeOp = vk::AttachmentStoreOp::eStore;
+    colorAttachmentInfos[0].loadOp = vk::AttachmentLoadOp::eLoad;
+    colorAttachmentInfos[0].clearValue.color = vk::ClearColorValue { .float32 = { { 0.0f, 0.0f, 0.0f, 0.0f } } };
+
+    vk::RenderingAttachmentInfoKHR depthAttachmentInfo {};
+    depthAttachmentInfo.imageView = resources->ImageResourceManager().Access(_gBuffers.Depth())->view;
+    depthAttachmentInfo.imageLayout = vk::ImageLayout::eDepthStencilAttachmentOptimal;
+    depthAttachmentInfo.storeOp = vk::AttachmentStoreOp::eDontCare;
+    depthAttachmentInfo.loadOp = vk::AttachmentLoadOp::eLoad;
+    depthAttachmentInfo.clearValue.depthStencil = vk::ClearDepthStencilValue { 1.0f, 0 };
+
+    vk::RenderingAttachmentInfoKHR stencilAttachmentInfo { depthAttachmentInfo };
+    stencilAttachmentInfo.storeOp = vk::AttachmentStoreOp::eDontCare;
+    stencilAttachmentInfo.loadOp = vk::AttachmentLoadOp::eDontCare;
+    stencilAttachmentInfo.clearValue.depthStencil = vk::ClearDepthStencilValue { 1.0f, 0 };
+
+    vk::RenderingInfoKHR renderingInfo {};
+    renderingInfo.renderArea.extent = vk::Extent2D { _gBuffers.Size().x, _gBuffers.Size().y };
+    renderingInfo.renderArea.offset = vk::Offset2D { 0, 0 };
+    renderingInfo.colorAttachmentCount = colorAttachmentInfos.size();
+    renderingInfo.pColorAttachments = colorAttachmentInfos.data();
+    renderingInfo.layerCount = 1;
+    renderingInfo.pDepthAttachment = &depthAttachmentInfo;
+    renderingInfo.pStencilAttachment = util::HasStencilComponent(_gBuffers.DepthFormat()) ? &stencilAttachmentInfo : nullptr;
 
-    // -- instanced rendering --
     util::BeginLabel(commandBuffer, "Particle rendering pass", glm::vec3 { 255.0f, 105.0f, 180.0f } / 255.0f, vkContext->Dldi());
+    commandBuffer.beginRenderingKHR(&renderingInfo, vkContext->Dldi());
+
+    commandBuffer.setViewport(0, 1, &_gBuffers.Viewport());
+    commandBuffer.setScissor(0, 1, &_gBuffers.Scissor());
 
     commandBuffer.bindPipeline(vk::PipelineBindPoint::eGraphics, _pipelines[static_cast<uint32_t>(ShaderStages::eRenderInstanced)]);
 
+    auto camera = scene.sceneDescription->camera;
+    glm::mat4 cameraRotation = mat4_cast(glm::quat(camera.eulerRotation));
+    glm::mat4 cameraTranslation = translate(glm::mat4 { 1.0f }, camera.position);
+    auto cameraView = glm::inverse(cameraTranslation * cameraRotation);
+    _renderPushConstant.cameraRight = glm::vec3(cameraView[0][0], cameraView[1][0], cameraView[2][0]);
+    _renderPushConstant.cameraUp = glm::vec3(cameraView[0][1], cameraView[1][1], cameraView[2][1]);
+    commandBuffer.pushConstants(_pipelineLayouts[static_cast<uint32_t>(ShaderStages::eRenderInstanced)], vk::ShaderStageFlagBits::eVertex, 0, sizeof(RenderPushConstant), &_renderPushConstant);
```

Comment:
use the modern overload of this function, something like
`commandBuffer.pushConstants<RenderPushConstant>(_pipelineLayouts[static_cast<uint32_t>(ShaderStages::eRenderInstanced)], vk::ShaderStageFlagBits::eVertex, 0, { _renderPushConstant });`